### DISCUSSION TITLE
Using Borg/Vorta to backup TrueNAS Core server to local machine?

### DISCUSSION THREAD

**Discussion by mikae1** [borgbackup/borg#7881] [Q&A]
Can I use Borg/Vorta to backup the TrueNAS Core server on my LAN to an external HDD attached to my desktop Linux machine?

I don't want to mess around with jails and install anything on the server. Considering this, is it possible? Has anybody found a good solution for this use case? Adding an SSH source in Vorta does not seem to be be possible. I guess I could mount the server using SSHFS and add it to sources, but is this the best way? 

Another thought is dropping the [borg-freebsd64](https://github.com/borgbackup/borg/releases/tag/1.2.6) binary in a dataset on the server and try to execute it from there and use the server as source and the external drive on my local machine as the destination (via SSH)

Borg is very interesting to me because it can compress the backups. My ZFS storage on the server uses zstd compression and so my backups will need to be compressed to fit on the external drive/drives connected to my desktop Linux machine.

**Comment by alsoeric** [comment]
you can. try these instructions. https://runby.coffee/backing-up-truenas-with-borg/

**Reply by mikae1** [reply]
> you can. try these instructions. https://runby.coffee/backing-up-truenas-with-borg/

I read that post, but as I wrote I don't want to install anything on the server. I will try a reverse procedure, like this:

borg create /run/media/USER/EXTERNALDRIVE::archive USER@IP:/mnt/BASEDATASET/

**Reply by ThomasWaldmann** [reply]
You can use borg in 2 ways:
- backup filesystem files to a directory (a borg repository)
- backup filesystem files to a ssh:// remote borg repository (needs borg on client and on server plus ssh)

As you dislike the 2nd way, you can use the 1st way if the machine running borg has fs access to both the (to be backed up) filesystem files (e.g. via a mounted network fs) and also to the borg repository (e.g. on a local disk, on a mounted usb disk, on a mounted network fs, ...).

**Comment by alsoeric** [comment]
> > you can. try these instructions. https://runby.coffee/backing-up-truenas-with-borg/
> 
> I read that post, but as I wrote I don't want to install anything on the server. 

I understand. I was reluctant to install anything on the server but since Borg would be installed in jail is much safer than installing in the native OS. I've used a separate VM to run borg and grab the backup source data from the NAS. There is a significant performance penalty especially if you're writing to another local storage target.

**Comment by mikae1** [comment]
> I understand. I was reluctant to install anything on the server but since Borg would be installed in jail is much safer than installing in the native OS. I've used a separate VM to run borg and grab the backup source data from the NAS. There is a significant performance penalty especially if you're writing to another local storage target.

Thanks, I've created a borg jail. Perhaps it would be preferred to used the self contained [borg-freebsd64](https://github.com/borgbackup/borg/releases/tag/1.2.6) binary? No dependencies to care about. If so, where should that executable go?

**Comment by ThomasWaldmann** [comment]
Closing this old thread - it was effectively answered in the comments (run borg on the TrueNAS side, e.g. in a jail or from the standalone FreeBSD binary, or pull the data to a machine that runs borg).

For the Vorta-specific parts of the question, Vorta is a separate project with its own tracker: https://github.com/borgbase/vorta

If you still have an open question about borg itself on a current version, please open a new discussion.
